xds/internal/balancer/cdsbalancer/cluster_handler_test.go

```diff
@@ -18,1078 +18,650 @@ package cdsbalancer
+				DNSHostname:      fmt.Sprintf("%s:%d", dnsHostNameNew, dnsPort),
+				OutlierDetection: json.RawMessage(`{}`),
+			},
+		},
+		XDSLBPolicy: json.RawMessage(`[{"xds_wrr_locality_experimental": {"childPolicy": [{"round_robin": {}}]}}]`),
 	}
-
-	// Invoke callback for Logical DNS child cluster.
-
-	fakeClient.InvokeWatchClusterCallback(xdsresource.ClusterUpdate{
-		ClusterType: xdsresource.ClusterTypeLogicalDNS,
-		ClusterName: logicalDNSService,
-	}, nil)
-
-	// Will Ping Cluster Handler, which will finally successfully build an
-	// update as all nodes in the tree of clusters have received an update.
-	// Since this cluster is an aggregate cluster comprised of two children, the
-	// returned update should be length 2, as the xds cluster resolver LB policy
-	// only cares about the full list of LogicalDNS and EDS clusters
-	// representing the base nodes of the tree of clusters. This list should be
-	// ordered as per the cluster update.
-	select {
-	case chu := <-ch.updateChannel:
-		if diff := cmp.Diff(chu.updates, []xdsresource.ClusterUpdate{{
-			ClusterType: xdsresource.ClusterTypeEDS,
-			ClusterName: edsService,
-		}, {
-			ClusterType: xdsresource.ClusterTypeLogicalDNS,
-			ClusterName: logicalDNSService,
-		}}); diff != "" {
-			t.Fatalf("got unexpected cluster update, diff (-got, +want): %v", diff)
-		}
-	case <-ctx.Done():
-		t.Fatal("Timed out waiting for the cluster update to be written to the update buffer.")
+	if err := compareLoadBalancingConfig(ctx, lbCfgCh, wantChildCfg); err != nil {
+		t.Fatal(err)
 	}
 }
 
-// TestUpdateRootClusterAggregateThenChangeChild tests the scenario where you
-// have an aggregate cluster with an EDS child and a LogicalDNS child, then you
-// change one of the children and send an update for the changed child. This
-// should write a new update to the update buffer to send back to CDS.
-func (s) TestUpdateRootClusterAggregateThenChangeChild(t *testing.T) {
-	// This initial code is the same as the test for the aggregate success case,
-	// except without validations. This will get this test to the point where it
-	// can change one of the children.
-	ch, fakeClient := setupTests()
-	ch.updateRootCluster(aggregateClusterService)
-
-	ctx, ctxCancel := context.WithTimeout(context.Background(), defaultTestTimeout)
-	defer ctxCancel()
-	_, err := fakeClient.WaitForWatchCluster(ctx)
-	if err != nil {
-		t.Fatalf("xdsClient.WatchCDS failed with error: %v", err)
+// Tests the case where the cluster resource requested by the cds LB policy is
+// an aggregate cluster root pointing to two child clusters, one of type EDS and
+// the other of type LogicalDNS. Verifies that the load balancing configuration
+// pushed to the cluster_resolver LB policy is as expected. The test then
+// updates the root cluster resource requested by the cds LB policy to a leaf
+// cluster of type EDS and verifies the load balancing configuration pushed to
+// the cluster_resolver LB policy.
```

Comment:
Nit: in this comment and many others, you mention "verifies the load balancing configuration". Can you please specify what exactly you're verifying in each case, or even just say verifies the load balancing configuration matches the leaf clusters or something like that.